### PR TITLE
Browser+LibGfx+LibWeb: Add dark mode

### PR DESCRIPTION
**Pull request by AtkinsSJ**
image

This PR adds:
- An 'IsDark' flag to system themes, which represents whether they should be considered a dark theme or not. (Along with support for flags in themes.)
- Support for the `@media (prefers-color-scheme: [light || dark]) {}` media query, which defaults to checking the theme's 'IsDark', but can be overridden with the `set_preferred_color_scheme()` IPC message.
- A 'Color Scheme' setting to Browser, which allows the user to set that override if they wish.

### CODE REVIEW

File: Userland/Libraries/LibGfx/SystemTheme.h [hunk 1]
```diff
@@ -132,6 +136,17 @@ inline const char* to_string(ColorRole role)
...
 
+enum class FlagRole {
+    NoRole,
+
+#undef __ENUMERATE_FLAG_ROLE
+#define __ENUMERATE_FLAG_ROLE(role) role,
+    ENUMERATE_FLAG_ROLES(__ENUMERATE_FLAG_ROLE)
+#undef __ENUMERATE_FLAG_ROLE
+
+        __Count,
```
Comment (linusg) [inline Userland/Libraries/LibGfx/SystemTheme.h:147]: This indentation looks really weird - sudden jump from 4 to 8 spaces?

File: Userland/Services/WebContent/PageHost.h [hunk 2]
```diff
@@ -75,6 +77,7 @@ class PageHost final : public Web::PageClient {
 
     RefPtr<Core::Timer> m_invalidation_coalescing_timer;
     Gfx::IntRect m_invalidation_rect;
+    String m_preferred_color_scheme { "auto" };
```
Comment (linusg) [inline Userland/Services/WebContent/PageHost.h:80]: Does this need to be a string? Does the IPC mechanism not have enums?

File: Userland/Libraries/LibGfx/SystemTheme.h (hunk 1 above)
Comment (AtkinsSJ) [reply Userland/Libraries/LibGfx/SystemTheme.h:147]: Clang-format does that for all the FooRole enums for some reason. :yakshrug:  ~~KlingFormat wen~~

File: Userland/Services/WebContent/PageHost.h (hunk 2 above)
Comment (AtkinsSJ) [reply Userland/Services/WebContent/PageHost.h]: Enums do work! I've switched to using one.

**Comment by AtkinsSJ** [comment]
Changes: Added a `PreferredColorScheme` enum that's used instead of passing Strings around.

**Comment by linusg** [comment]
Much nicer! Let's get this merged :^)

**Pull request merged**